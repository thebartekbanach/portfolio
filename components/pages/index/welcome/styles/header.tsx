import styled from "styled-components";
import { useFont } from "~/utils/styles/fonts";
import { textGradient } from "~/utils/styles/textGradient";

export const HeaderContainer = styled.div`
	padding-top: 120px;

	@media (max-width: 1150px) {
		padding-top: 100px;
	}

	@media (max-width: 800px) {
		padding-top: 20px;
		padding-bottom: 20px;
	}
`;

export const NameHeader = styled.h1`
	${useFont.nunitoSans.bold};
	font-size: 75px;
	letter-spacing: 2px;
	margin-bottom: 10px;
	margin-left: -5px;
	margin-top: 0;

	${textGradient("to top", "#4764e6", "#694fff", "#4764e6")}

	@media (max-width: 1150px) {
		font-size: 65px;
	}

	@media (max-width: 800px) {
		font-size: 11vw;
	}

	@media (max-width: 300px) {
		font-size: 45px;
	}
`;

export const SpecializationSubheader = styled.h2`
	margin-top: 10px;
	margin-bottom: 20px;

	${useFont.nunitoSans.semiBold};
	font-size: 30px;
	letter-spacing: 1px;
	color: #4764e6;
	margin-left: -2px;

	@media (max-width: 800px) {
		font-size: 5.5vw;
	}

	@media (max-width: 300px) {
		font-size: 20px;
	}
`;
