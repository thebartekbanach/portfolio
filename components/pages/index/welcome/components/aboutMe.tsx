import { FC } from "react";
import { AboutMeContainer } from "../styles/aboutMe";

interface AboutMeProps {
	content: string;
}

export const AboutMe: FC<AboutMeProps> = ({ content }) => (
	<AboutMeContainer dangerouslySetInnerHTML={{ __html: content }} />
);
