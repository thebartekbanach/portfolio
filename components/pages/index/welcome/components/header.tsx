import { FC } from "react";
import { NameHeader, SpecializationSubheader, HeaderContainer } from "../styles/header";

interface HeaderProps {
	name: string;
	specialization: string;
}

export const Header: FC<HeaderProps> = ({ name, specialization }) => (
	<HeaderContainer>
		<NameHeader>{name}</NameHeader>
		<SpecializationSubheader>{specialization}</SpecializationSubheader>
	</HeaderContainer>
);
