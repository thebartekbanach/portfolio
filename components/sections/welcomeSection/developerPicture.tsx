import { DeveloperPictureContainer, DeveloperImage } from "./styles/developerPicture";

export const DeveloperPicture = () => (
	<DeveloperPictureContainer>
		<DeveloperImage src="/img/svg/developerPicture.svg" />
	</DeveloperPictureContainer>
);
