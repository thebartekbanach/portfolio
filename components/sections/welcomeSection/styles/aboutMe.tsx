import styled from "styled-components";
import { useFont } from "~/utils/styles/fonts";
import { textGradient } from "~/utils/styles/textGradient";

export const AboutMeContainer = styled.div`
	max-width: 40%;

	${useFont.nunitoSans.regular}
	font-size: 18px;
	color: #5a5f69;

	line-height: 1.4em;
	letter-spacing: 0.02em;

	@media (max-width: 1000px) {
		max-width: 45%;
	}

	@media (max-width: 800px) {
		max-width: 100%;
		margin-top: 50px;
	}

	b {
		${textGradient("to top", "#4764e6", "#694fff", "#4764e6")}
	}
`;
