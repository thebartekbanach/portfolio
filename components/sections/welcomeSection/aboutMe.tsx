import { FC } from "react";
import { AboutMeContainer } from "./styles/aboutMe";

interface AboutMeProps {
	content: string;
}

export const AboutMe: FC<AboutMeProps> = ({ content }) => (
	<AboutMeContainer lang="pl" dangerouslySetInnerHTML={{ __html: content }} />
);
