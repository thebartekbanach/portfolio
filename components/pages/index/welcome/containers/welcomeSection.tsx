import { Navbar } from "~/components/shared/navbar/containers/navbar";
import { Section } from "~/components/shared/layout/components/section";
import { Header } from "~/components/pages/index/welcome/components/header";
import { WelcomeSection as WelcomeSectionContainer } from "~/components/pages/index/welcome/components/container";
import { AboutMe } from "~/components/pages/index/welcome/components/aboutMe";
import { DeveloperPicture } from "~/components/pages/index/welcome/components/developerPicture";
import { useTranslation } from "~/utils/i18next";

export const WelcomeSection = () => {
	const { t } = useTranslation("welcome");

	return (
		<Section isFirstSection>
			<Navbar />
			<WelcomeSectionContainer>
				<Header name={t("name")} specialization={t("specialization")} />
				<DeveloperPicture />
				<AboutMe content={t("aboutMe")} />
			</WelcomeSectionContainer>
		</Section>
	);
};
