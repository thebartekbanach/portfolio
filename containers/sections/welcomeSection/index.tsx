import { Navbar } from "~/containers/Navbar";
import { Section } from "~/components/shared/layout/section";
import { Header } from "~/components/sections/welcomeSection/header";
import {
	WelcomeSectionContentContainer,
	WelcomeSectionContent
} from "~/components/sections/welcomeSection/container";
import { AboutMe } from "~/components/sections/welcomeSection/aboutMe";
import { DeveloperPicture } from "~/components/sections/welcomeSection/developerPicture";

export const WelcomeSection = () => {
	const aboutMe = `
		Jestem <b>20 letnim</b> w pełni samodzielnym <b>pasjonatem web developmentu</b>.
		<b>Kocham automatyzować zadania, ułatwiać pracę ludziom</b> nie tylko
		za pomocą aplikacji internetowych ale także za pomocą intuicyjnych
		systemów typu embedded. Lubię też <b>tworzyć proste, intuicyjne
		aplikacje frontendowe</b>, które są proste w użyciu oraz przydatne w codziennym życiu.
	`;

	return (
		<Section>
			<Navbar />
			<WelcomeSectionContentContainer>
				<WelcomeSectionContent>
					<Header name="Bartek Banach" specialization="Full Stack Developer" />
					<DeveloperPicture />
					<AboutMe content={aboutMe} />
				</WelcomeSectionContent>
			</WelcomeSectionContentContainer>
		</Section>
	);
};
