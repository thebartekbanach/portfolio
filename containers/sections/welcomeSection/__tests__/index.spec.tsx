jest.mock("~/utils/i18next");

import { WelcomeSection } from "..";
import { shallow } from "enzyme";
import { useTranslationMock } from "~/tests/utils/useTranslationMock";
import { Header } from "~/components/sections/welcomeSection/header";
import { DeveloperPicture } from "~/components/sections/welcomeSection/developerPicture";
import { AboutMe } from "~/components/sections/welcomeSection/aboutMe";

describe(WelcomeSection, () => {
	it("should match snapshot", () => {
		useTranslationMock("welcome", {
			name: "Bartek Banach",
			specialization: "Full Stack Developer",
			aboutMe: "This is some description about me"
		});

		const result = shallow(<WelcomeSection />);

		expect(result).toMatchSnapshot();
	});

	it("should match snapshot", () => {
		useTranslationMock("welcome", {
			name: "Bartek Banach",
			specialization: "Full Stack Developer",
			aboutMe: "This is some description about me"
		});

		const rendered = shallow(<WelcomeSection />);
		const result = rendered.containsAllMatchingElements([
			<Header key="1" name="Bartek Banach" specialization="Full Stack Developer" />,
			<DeveloperPicture key="2" />,
			<AboutMe key="3" content="This is some description about me" />
		]);

		expect(result).toBeTruthy();
	});
});
