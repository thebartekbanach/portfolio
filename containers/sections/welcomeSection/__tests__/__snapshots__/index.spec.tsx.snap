// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSection should match snapshot 1`] = `
<section__Section>
  <Navbar />
  <container__WelcomeSection>
    <Header
      name="Bartek Banach"
      specialization="Full Stack Developer"
    />
    <DeveloperPicture />
    <AboutMe
      content="This is some description about me"
    />
  </container__WelcomeSection>
</section__Section>
`;
